cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/StreamServiceRuntimeModule.java

```diff
@@ -73,6 +74,7 @@ protected void configure() {
         bind(StreamFileJanitorService.class).to(DistributedStreamFileJanitorService.class).in(Scopes.SINGLETON);
         bind(StreamWriterSizeCollector.class).to(BasicStreamWriterSizeCollector.class).in(Scopes.SINGLETON);
         bind(StreamService.class).to(DistributedStreamService.class).in(Scopes.SINGLETON);
+        bind(AbstractNamespaceClient.class).to(DiscoveryNamespaceClient.class).in(Scopes.SINGLETON);
```

Comment:
If there is, I'm not sure how. We can put that one binding in a separate module, and use that module. I'm not sure thats necessary though - it'd be a very tiny module, and would still need to be repeated across the three modes.